### PR TITLE
Added: search language support

### PR DESCRIPTION
**Pull request by upupming**
I added Chinese language support for Algolia and local search, which take advantage of  the yml files under the language folder.  In this way, it will be much more extensible.

screenshot from 2018-06-21 12-43-38
screenshot from 2018-06-21 12-42-09

### CODE REVIEW

File: layout/includes/config.pug [hunk 1]
```diff
@@ -9,16 +11,22 @@
...
-  } else {
-    algolia = 'undefined'
-  }
+    algolia_search = JSON.stringify({
+      input_placeholder: _p("algolia_search.input_placeholder"),
+      hits_empty: _p("algolia_search.hits_empty"),
+      hits_stats: _p("algolia_search.hits_stats")
+    })
+  } 
   var localSearch = 'undefined';
```
Comment (Molunerfinn) [inline layout/includes/config.pug]: There is a `localSearch`

File: layout/includes/config.pug [hunk 2]
```diff
@@ -9,16 +11,22 @@
...
+  // search languages
+  var local_search = 'undefined';
```
Comment (Molunerfinn) [inline layout/includes/config.pug]: So this `local_search` is redundant. Can you just use one of it?

**Comment by upupming** [comment]
Thanks for your review, I will fix that soon!

**Comment by Molunerfinn** [comment]
Thanks for your contribution!

**Pull request merged**